mvpa2/cmdline/cmd_ttest.py

```diff
@@ -58,14 +61,38 @@ def setup_parser(parser):
                         belong to.  E.g. if values are accuracies, it would be the
                         'greater', if errors -- the 'less'""")
 
+def guess_backend(fn):
+    if fn.endswith('.gz'):
```

Comment:
How about an abstract class that has a static method guess_backend, and concrete Backend_* classes that provide methods for loading (e.g. nib.load) and getting data (e.g .get_data)? Possibly also a method indicating which extensions are supported.